### DISCUSSION TITLE
How to connect with only user and password

### DISCUSSION THREAD

**Discussion by freddant** [FreeOpcUa/opcua-asyncio#1602] [Q&A]
Hi there, been trying to connect to opc ua server that is configured to accept connections with only user and password, but no success.
The server is configured as in the picture.
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/74458790/e65f18dc-bc18-4c2c-8bab-4536ebe5255e)
The code i'm using is

```
    opcua_client = Client(OPCUA_ADDRESS)
    opcua_client.set_user(opcua_user)
    opcua_client.set_password(opcua_pass)
    opcua_client.set_security_string("Basic256Sha256,SignAndEncrypt")

    try:
        opcua_client.connect()
        root = opcua_client.get_root_node()
        print(root)

        print(root.get_children())


    finally:
        opcua_client.disconnect()
```

But i get a timeout error.
What am i doing wrong?

**Comment by AndreasHeine** [comment]
make sure your server url is correct and you need a x509v3 certificate and private key in order to use sign and encrypt!

e.g. 
https://github.com/AndreasHeine/SecurePythonOpcUaClient/blob/master/x509v3%2Fclient_x509v3.py

**Comment by freddant** [comment]
Hey, thanks for the fast reply, can you clarify something for me? With that server configuration (the one from the image) is it possible to connect without the certificate and the private key (only with username and password)? Sorry if the question sounds redundant. 
Cheers!

**Reply by AndreasHeine** [reply]
no! you need the key and cert for the encrypted connection (transport security)

username and pw is the authentification method 

there is certificate auth aswell in opc ua but transport and auth are two topics


reference:
https://reference.opcfoundation.org/Core/Part2/v104/docs/8

**Reply by freddant** [reply]
The certificate and private key needs to be on the server-side only or is it necessary for the client to connect as well?

**Reply by AndreasHeine** [reply]
opc ua spec says each opc ua application! means client aswell ;)

**Comment by liuyazui** [comment]
@freddant hi, do you success? I have the same problem
